mne/preprocessing/ica.py

```diff
@@ -1029,6 +1036,102 @@ def _check_target(self, target, inst, start, stop,
 
         return target
 
+    def _find_bads_ch(self, inst, chs, threshold=3.0, start=None,
+                      stop=None, l_freq=None, h_freq=None,
+                      reject_by_annotation=True, verbose=None, prefix="chs"):
+        """Help function for eog_find_bads_eog/ecg/ref.
+
+        Detection is based on Pearson correlation between the MEG data
+        components and MEG reference components.
+        Thresholding is based on adaptive z-scoring. The above threshold
+        components will be masked and the z-score will be recomputed
+        until no supra-threshold component remains.
+
+        Parameters
+        ----------
+        inst : instance of Raw, Epochs or Evoked
+            Object to compute sources from. Should contain at least one channel
+            i.e. component derived from MEG reference channels.
+        chs: list of array-like | list of ch_names
+            Which data/channels to use as targets
+        threshold : int | float
+            The value above which a feature is classified as outlier.
+        start : int | float | None
+            First sample to include. If float, data will be interpreted as
+            time in seconds. If None, data will be used from the first sample.
+        stop : int | float | None
+            Last sample to not include. If float, data will be interpreted as
+            time in seconds. If None, data will be used to the last sample.
+        l_freq : float
+            Low pass frequency.
+        h_freq : float
+            High pass frequency.
+        reject_by_annotation : bool
+            If True, data annotated as bad will be omitted. Defaults to True.
+
+            .. versionadded:: 0.14.0
+
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see
+            :func:`mne.verbose` and :ref:`Logging documentation <tut_logging>`
+            for more). Defaults to self.verbose.
+
+        Returns
+        -------
+        ref_idx : list of int
+            The indices of channel related ICA components, sorted by score.
+        scores : np.ndarray of float, shape (``n_components_``) | list of array
+            The correlation scores.
+
+        See Also
+        --------
+        find_bads_ecg, find_bads_eog, _find_bads_ch
```

Comment:
FYI I have stopped documenting private functions like this and instead do:
```
"""Compute ExG/ref components.

See find_bads_ecg, find_bads, eog, and find_bads_ref for details.
"""
```
This makes the code a bit easier to read and maintain in the end because things are more DRY. Otherwise the docstrings end up often out of sync anyway, and do more harm than good.